src/rust/src/file_reader.rs

```diff
@@ -497,20 +535,21 @@ impl<R: Read, S> FeatureIter<R, S> {
             let _feature = size_prefixed_root_as_feature(&self.fbs.feature_buf)?;
         }
         self.feat_no += 1;
-        self.cur_pos += self.fbs.feature_buf.len() as u64;
```

Comment:
feature_buf is the size_prefixed_root_as_feature. So it includes the size_prefix which is 4 bytes + feature_size.

In the new code, we *only* add feature_size here since we've already accounted for the +4 bytes closer to the actual reading, in read_feature_size.